spark/src/main/kotlin/com/adevinta/spark/res/AnnotatedStringResource.kt

```diff
@@ -0,0 +1,238 @@
+package com.adevinta.spark.res
+
+import android.content.res.Resources
+import android.graphics.Typeface
+import android.text.Annotation
+import android.text.Spanned
+import android.text.SpannedString
+import android.text.style.AbsoluteSizeSpan
+import android.text.style.BulletSpan
+import android.text.style.ForegroundColorSpan
+import android.text.style.RelativeSizeSpan
+import android.text.style.StrikethroughSpan
+import android.text.style.StyleSpan
+import android.text.style.SubscriptSpan
+import android.text.style.SuperscriptSpan
+import android.text.style.TypefaceSpan
+import android.text.style.UnderlineSpan
+import android.util.Log
+import androidx.annotation.StringRes
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.BaselineShift
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.em
+import androidx.core.text.HtmlCompat
+import com.adevinta.spark.PreviewTheme
+import com.adevinta.spark.R
+import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.tokens.SparkColors
+import com.adevinta.spark.tokens.SparkTypography
+
+
+// FIXME: There is no official way to do this yet so we're waiting for
+//   https://issuetracker.google.com/issues/139320238 to be fixed
+@Composable
+@ReadOnlyComposable
+public fun resources(): Resources {
+    LocalConfiguration.current
+    return LocalContext.current.resources
+}
+
+public fun Spanned.toHtmlWithoutParagraphs(): String {
+    return HtmlCompat.toHtml(this, HtmlCompat.TO_HTML_PARAGRAPH_LINES_CONSECUTIVE)
+        .substringAfter("<p dir=\"ltr\">").substringBeforeLast("</p>")
+}
+
+public fun Resources.getText(@StringRes id: Int, vararg args: Any): CharSequence {
+    val escapedArgs = args.map {
+        if (it is Spanned) it.toHtmlWithoutParagraphs() else it
+    }.toTypedArray()
+    val resource = SpannedString(getText(id))
+    val htmlResource = resource.toHtmlWithoutParagraphs()
+    val formattedHtml = String.format(htmlResource, *escapedArgs)
+    return HtmlCompat.fromHtml(formattedHtml, HtmlCompat.FROM_HTML_MODE_LEGACY)
+}
+
+@Composable
+public fun annotatedStringResource(@StringRes id: Int, vararg formatArgs: Any): AnnotatedString {
```

Comment:
The naming of these functions do not represent their return type. They don't return a resource but an annotated string, this feels weird to me.

Reply:
Actually we're following the naming for ressources from Compose like these api:
``` kotlin
stringResource(id = )
stringArrayResource(id = )
pluralStringResource(id = , count = )
imageResource(id= )
```